plugins/SitesManager/API.php

```diff
@@ -1346,6 +1357,43 @@ public function setDefaultTimezone($defaultTimezone)
         return true;
     }
 
+    public function setExclusionTypeForQueryParams(string $exclusionTypeForQueryParams): void
+    {
+        Piwik::checkUserHasSuperUserAccess();
+
+        if (!in_array($exclusionTypeForQueryParams, SitesManager::URL_PARAM_EXCLUSION_TYPES)) {
+            throw new Exception($this->translator->translate('SitesManager_ExceptionInvalidExclusionType'));
+        }
```

Comment:
No need to have a custom check and error message for this. You could use the `WhitelistedValue` validator to check that.

Reply:
I didn't know about that, I've updated to use it let me know if that's what you meant